script.sh

```diff
@@ -1,9 +1,75 @@
-#!/bin/sh
+#!/bin/bash
 set -
+
+DOWNLOAD_URL="https://www.supermicro.com/wdl/utility/IPMIView/Linux/"
+LOCAL_DOWNLOAD_LOCATION="./SM_download"
+
+if which wget >/dev/null; then
+  echo "Downloading latest version of IPMIView from [${DOWNLOAD_URL}]..."
+  wget \
+    --timestamping \
+    --recursive \
+    --level=1 \
+    -q \
+    --show-progress \
+    --directory-prefix="${LOCAL_DOWNLOAD_LOCATION}/" \
+    --no-parent \
+    --no-directories \
+    --reject index.html,index.html.tmp,robots.txt,robots.txt.tmp \
+    "${DOWNLOAD_URL}"
+  rm "${LOCAL_DOWNLOAD_LOCATION}/"robots.txt*
+
+  # Check MD5
+  EXPECTED_MD5=$(\grep MD5 "${LOCAL_DOWNLOAD_LOCATION}/CheckSum.txt" | cut -d':' -f2 | tr -d "[:space:]" | tr '[:upper:]' '[:lower:]')
+  ACTUAL_MD5=$(md5sum "${LOCAL_DOWNLOAD_LOCATION}"/IPMIView*.tar* | cut -d' ' -f1 | tr -d "[:space:]" | tr '[:upper:]' '[:lower:]')
+  if ! diff <(echo "${EXPECTED_MD5}") <(echo "${ACTUAL_MD5}"); then
+    echo "MD5 is not as expected; download corrupted."
+    echo "Expected: [${EXPECTED_MD5}]"
+    echo "Actual:   [${ACTUAL_MD5}]"
+    echo "Exiting."
+    exit 1
+  fi
+
+  # Check SHA-256
+  EXPECTED_SHA256=$(\grep MD5 "${LOCAL_DOWNLOAD_LOCATION}/CheckSum.txt" | cut -d':' -f2 | tr -d "[:space:]" | tr '[:upper:]' '[:lower:]')
+  ACTUAL_SHA256=$(md5sum "${LOCAL_DOWNLOAD_LOCATION}"/IPMIView*.tar* | cut -d' ' -f1 | tr -d "[:space:]" | tr '[:upper:]' '[:lower:]')
+  if ! diff <(echo "${EXPECTED_SHA256}") <(echo "${ACTUAL_SHA256}"); then
+    echo "SHA-256 is not as expected; download corrupted."
+    echo "Expected: [${EXPECTED_SHA256}]"
+    echo "Actual:   [${ACTUAL_SHA256}]"
+    echo "Exiting."
+    exit 1
+  fi
+
+else
+  echo "WARNING: 'wget' CLI not found."
+  echo
+  echo "Please visit ${DOWNLOAD_URL} to download the latest version of IPMIView and copy the archive into $(pwd)/${LOCAL_DOWNLOAD_LOCATION}/"
+  echo
+  # shellcheck disable=SC2034,SC2162
+  echo "Press [Enter] to continue" && read answer
+fi
+
+echo "Extracting contents of downloaded IPMIView archive..."
+if [[ -d Contents/ ]]; then
+  rm -rf Contents/
```

Comment:
This deletes the MacOS directory that holds the app run rules.  As such, the app will not run.

Reply:
Nice, thanks for catching this! It was working totally fine for me locally, I suspect because I was not deleting the app between runs (ie, `rm -rf ~/Applications/IPMIView.app`) and the script just rsync's changes, so the required files were already in my local directory. I'm not sure if we'd want to add `--delete` to the `rsync` command, but something we might want to consider so that subsequent runs between versions don't leave any lingering files. For the time being, I think it's fine without doing that.

One thought, I wonder if it might be good for us to have a script for 'developer testing' that does the necessary purging (ie, `rm -rf ~/Applications/IPMIView.app`) and attempts to run the app afterwards (ie, `open ~/Applications/IPMIView.app`). Something as simple as:
```bash
rm -rf ~/Applications/IPMIView.app
bash script.sh
open ~/Applications/IPMIView.app
```